packages/amplify-console-integration/__tests__/pullAndInit.test.ts

```diff
@@ -0,0 +1,156 @@
+import * as fs from 'fs-extra';
+import {
+  getConfiguredAmplifyClient,
+  createConsoleApp,
+  generateBackendEnvParams,
+  createBackendEnvironment,
+  deleteAllAmplifyProjects,
+} from '../src/pullAndInit/amplifyConsoleOperations';
+
+import { headlessInit } from '../src/pullAndInit/initProject';
+import { headlessPull } from '../src/pullAndInit/pullProject';
+import { headlessDelete } from '../src/pullAndInit/deleteProject';
+import {
+  removeFilesForTeam,
+  removeFilesForThirdParty,
+  checkAmplifyFolderStructure,
+  getTeamProviderInfo,
+} from '../src/pullAndInit/amplifyArtifactsManager';
+import * as util from '../src/util';
+
+describe('amplify console build', () => {
+  beforeAll(() => {
+    return deleteAllAmplifyProjects();
+  });
+
+  afterAll(() => {
+    return deleteAllAmplifyProjects();
+  });
+
+  beforeEach(() => {});
+
+  afterEach(async () => {});
+
+  it('connects to team project', async () => {
+    const amplifyClient = getConfiguredAmplifyClient();
+    const projectName = 'cliintegrationteam';
+
+    const appId = await createConsoleApp(projectName, amplifyClient);
+    let envName = 'enva';
+    let backendParams = generateBackendEnvParams(appId, projectName, envName);
+
+    await createBackendEnvironment(backendParams, amplifyClient);
+
+    let amplifyParam = {
+      envName,
+      appId,
+    };
+    const providersParam = {
+      awscloudformation: {
+        configLevel: 'project',
+        useProfile: true,
+        profileName: util.getProfileName(),
+      },
+    };
+    const codegenParam = {
+      generateCode: false,
+      generateDocs: false,
+    };
+
+    let teamProviderInfo;
+    //test for init a clean frontend project
+    const projectDirPath = util.createNewProjectDir();
+    await headlessInit(projectDirPath, amplifyParam, providersParam, codegenParam);
+    expect(checkAmplifyFolderStructure(projectDirPath)).toBeTruthy();
+    teamProviderInfo = getTeamProviderInfo(projectDirPath);
+    expect(teamProviderInfo).toBeDefined();
+
+    //test for existing env
+    removeFilesForTeam(projectDirPath);
+    await headlessPull(projectDirPath, amplifyParam, providersParam);
+    await headlessInit(projectDirPath, amplifyParam, providersParam, codegenParam);
+    expect(checkAmplifyFolderStructure(projectDirPath)).toBeTruthy();
+    teamProviderInfo = getTeamProviderInfo(projectDirPath);
+    expect(teamProviderInfo).toBeDefined();
+    expect(teamProviderInfo['enva']).toBeDefined();
```

Comment:
@UnleashedMind Would it make more sense to use the envName var instead?

```suggestion
    expect(teamProviderInfo[envName]).toBeDefined();
```

Reply:
we are testing two scenarios, one for a single new env, which will be used as the base for comparison, hence it's named "enva" and the other environment is named "envb"